keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -178,21 +176,16 @@ class SqlArtifactRepository(
   override fun latestVersionApprovedIn(
     deliveryConfig: DeliveryConfig,
     artifact: DeliveryArtifact,
-    targetEnvironment: String,
-    statuses: List<ArtifactStatus>
+    targetEnvironment: String
   ): String? {
     val environment = deliveryConfig.environmentNamed(targetEnvironment)
     val envUid = deliveryConfig.getUidFor(environment)
     val artifactId = artifact.uid
     val versions: List<String> = jooq
       .select(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION)
-      .from(ENVIRONMENT_ARTIFACT_VERSIONS, ARTIFACT_VERSIONS)
+      .from(ENVIRONMENT_ARTIFACT_VERSIONS)
       .where(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(envUid))
       .and(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artifactId))
```

Comment:
shouldn't this also have something like: 

`.and(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS.eq(PromotionStatus.APPROVED.toString()))`

Reply:
I think yes! I'm going to dig into the logic here to understand why it wasn't like that before, and return with a fix or an explanation.

Reply:
Ok, so we don't write to this table unless the artifact is approved. Basically any `PromotionStatus` means it's been approved for the environment. So we don't need that.